File: report_app/lib/model/login_model.dart
import 'dart:convert';

import 'package:http/http.dart' as http;
import 'package:report_app/data/error_response.dart';
import 'package:report_app/global.dart';
import 'package:shared_preferences/shared_preferences.dart';

import '../data/account.dart';

class LoginModel {
  Future<Account> login(String email, String password) async {
    var response = await http.post(Uri.parse("${baseUrl}login"),
        headers: {
          'Content-type': 'application/json'
        },
        body: utf8.encode(jsonEncode({
          'email': email,
          'password': password
        }))
    ).timeout(const Duration(seconds: 10));

    if(response.statusCode != 200) {
      var error = ErrorResponse.fromJson(jsonDecode(response.body));
      throw Exception(error.error);
    }
    var account = Account.fromJson(jsonDecode(response.body));
    return account;
  }

  Future<void> saveLoginInfo(Account account) async {
    await SharedPreferences.getInstance()
      ..setInt("id", account.id)
      ..setString("name", account.name)
      ..setString("email", account.email)
      ..setString("password", account.password)
      ..setBool("isLogin", true);
  }
}

File: Old Version/ReportApp_Android/app/src/main/java/com/seielika/reportapp/Model/LoginModel.kt
package com.seielika.reportapp.Model

import android.content.Context
import androidx.core.content.edit
import com.google.gson.Gson
import com.seielika.reportapp.Data.Login
import com.seielika.reportapp.Data.LoginResponse
import com.seielika.reportapp.Global
import okhttp3.*
import okhttp3.RequestBody.Companion.toRequestBody
import java.io.IOException
import java.time.Duration

class LoginModel {
    fun login(email: String, password: String, callBack: Callback) {
        val body = Login(email, password)
        val requestBody = Gson().toJson(body).toRequestBody()

        val client = OkHttpClient.Builder()
            .connectTimeout(Duration.ofSeconds(10))
            .build()

        val request = Request.Builder()
            .method("POST", requestBody)
            .header("Content-Type", "application/json")
            .url(Global.baseURL + "login")
            .build()

        client.newCall(request).enqueue(callBack)
    }

    fun saveLoginInfo(context: Context, login: LoginResponse) {
        val sharedPreferences = context.getSharedPreferences("User", Context.MODE_PRIVATE)
        sharedPreferences.edit {
            putInt("id", login.id)
            putString("name", login.name)
            putString("email", login.email)
            putString("password", login.password)
            putBoolean("isLogin", true)
        }
    }
}